[jamz903] iP

jamz903 opened the pull request:
### OrangesBot

> “Orange is the happiest color.” – Frank Sinatra ([source](https://everydaypower.com/orange-quotes/))

OrangesBot frees your mind of having to remember things you need to do. It's fast and easy, just like peeling an orange. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/jamz903/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

------
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jamz903 committed:
Level-0

jamz903 committed:
Level-1

jamz903 committed:
Level-2

jamz903 committed:
level-3

jamz903 committed:
level-4

jamz903 committed:
A-TextUiTesting

jamz903 committed:
Level-5

jamz903 committed:
Level-6

jamz903 committed:
Add ability to save and recall from hard disk

jamz903 committed:
Store deadlines and events in datetime format

jamz903 committed:
Refactor code into Ui, Storage, Parser and TaskList classes

jamz903 committed:
Minor Bug Fixes

jamz903 committed:
More Minor Bug Fixes & Nest into Package Duke

jamz903 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

jamz903 committed:
Set up Gradle and add JUnit Tests

jamz903 committed:
Add JavaDoc to Deadline, Duke, Event, Parser files

jamz903 committed:
Comply with coding standard

jamz903 committed:
Merge branch 'branch-A-CodingStandard'

jamz903 committed:
Add find task

jamz903 committed:
Merge branch 'branch-Level-9'

m1oojv commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,66 @@
+package duke;
+
+import java.io.*;
```
Cannot use import wildcard. Do list down what files from java.io is being used. 
I think this mistake also happened across a few of your files too.
eg. `import java.io.File`

m1oojv reviewed (commented):
Overall, I found your code easy to read for the most part. I noted a few minor coding standard violations.

richiehx commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,136 @@
+        }
+        else if (promptText.startsWith("deadline")) {
+            try {
+                String[] parts = promptText.split("/", 2);
```
Perhaps you could use a more descriptive variable name to help with understanding.

richiehx commented on src/main/java/duke/Parser.java:
Perhaps you could move the creation of tasks into the TaskList and have the Parser only contain the parsing components?

richiehx commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,68 @@
+import java.util.Scanner;
+
+public class Ui {
+    private static Scanner prompt = new Scanner(System.in);
```
Nice variable names! 👍

richiehx commented on src/main/java/duke/DukeException.java:
You might want to move the Exception into it's own package (eg. exceptions) to help increase visibility of the separation of code logic!

richiehx reviewed (commented):
LGTM! Maybe you could separate your files into different packages.

seanpzk commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,136 @@
+                throw new DukeException("OOPS!! The description of an event cannot be empty.");
+            }
+        }
+    }
```
Good job on implementation of createTask, the flow of the code makes sense (how we handle todo, then deadline and lastly event). Perhaps, the formatting of date could be handled by a separate function or class to make the code more readable!

seanpzk commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,68 @@
+        echo(tasks);
+    }
+
+    public static void deleteTask(TaskList tasks, int i) throws DukeException {
```
I like how this method makes use of DukeException to inform the user that the task to be deleted does not exist. As this function deals with deletion of task, perhaps it could be a method implemented under TaskList.java.

seanpzk commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,68 @@
+            tasks.delete(tasks.get(i));
+        }
+        catch (IndexOutOfBoundsException e) {
+            throw new DukeException("OOPS!! This task doesn't exist!");
```
As mentioned earlier, this makes good use of DukeException. Maybe it would be better to relate the error message to the IndexOutOfBoundsException.
```suggestion
            throw new DukeException("OOPS!! There is only " + tasks.size() + " tasks. Please input a number up till " + tasks.size() );
```

seanpzk commented on src/test/java/ParserTest.java:
```diff
@@ -0,0 +1,127 @@
+    }
+
+
+}
```
Good effort on implementing the test cases to test different date formats. Perhaps the test name can include something about date format (rather than dash, slash and others) to make it clear that test focuses on testing date formats.

seanpzk commented on src/test/java/StorageTest.java:
```diff
@@ -0,0 +1,49 @@
+        assertEquals(taskList.get(0), actualList.get(0));
+    }
+
+}
```
Great work on the test cases, they cover important functionalities of Storage to ensure that the class is working well!

jamz903 committed:
Add GUI and fix minor mistakes

jamz903 committed:
Remove redundant code

Chen-Kuei commented on src/main/java/duke/Ui.java:
```diff
@@ -1,13 +1,9 @@
 package duke;
 
 public class Ui {
-    private Duke duke;
```
LGTM! It's very good to have this part (remove redundant code), which makes your whole program cleaner and easy to read. :blush:

Chen-Kuei commented on src/main/java/List.java:
```diff
@@ -0,0 +1,18 @@
+import java.util.ArrayList;
+
+public class List {
```
Good job for creating the List class here, so your program is more OOP!

Chen-Kuei commented on src/main/java/Duke.java:
```diff
@@ -29,41 +37,64 @@ else if (promptText.startsWith("mark") || promptText.startsWith("unmark")){
+    public static void markTask(String promptText) throws DukeException {
+        try {
+            int i = Integer.parseInt(promptText.substring(promptText.length() - 1));
+            Task t = taskList.get(i-1);
```
Maybe the naming of variables can be improved? Like 'int i' can be changed to 'int index' and can use 'Task task' instead of 'Task t', so that it will be easier for the readers to understand.

Chen-Kuei reviewed (commented):
**Looks good**. Great job on this!! :thumbsup:

jamz903 committed:
Add Assertions to Document Assumptions

Incorporate Java Assertions to document crucial assumptions throughout the codebase.

This change promotes better code quality and ensures that essential assumptions are explicitly documented and enforced.

jamz903 committed:
Refactor Code and Abstract Functionality into DateParser Class

This commit involves a significant code refactoring effort aimed at enhancing code quality and maintainability.

Specifically, portions of logic in Parser has been abstracted out to DateParser, a dedicated class to process the Date objects of the Tasks.

This change makes it easier to maintain and sets the stage for more efficient development and easier debugging in subsequent iterations.

jamz903 committed:
Support recording info about places

This commit introduces a new feature for managing information about places, such as details about Food, Shopping or Study places that users have visited. This functionality enables users to record information for future reference and convenience.

jamz903 committed:
Merge pull request #1 from jamz903/branch-A-Assertions

Add Assertions to Document Assumptions

jamz903 committed:
Merge pull request #2 from jamz903/branch-A-CodeQuality

Refactor Code and Abstract Functionality into DateParser Class

jamz903 committed:
Merge pull request #3 from jamz903/branch-BCD-Extension

Support recording info about places

jamz903 committed:
Improve code quality

jamz903 committed:
Minor Bug Fix

jamz903 committed:
Add User Guide
